src/main/kotlin/com/manimdsl/frontend/Type.kt

```diff
@@ -160,10 +167,11 @@ data class TreeType(
 
 data class NodeType(
     override var internalType: Type,
+) : DataStructureType(internalType), NullableDataStructure {
     override val methods: MutableMap<String, DataStructureMethod> = hashMapOf(
-        "left" to Left(internalType), "right" to Right(internalType), "value" to Value(internalType)
-    ),
-) : DataStructureType(internalType, methods), NullableDataStructure {
+        "left" to Left(this), "right" to Right(this), "value" to Value(internalType)
```

Comment:
This makes it a recursive datatype so checking the access chain is semantically correct now as before it would just check each access with the identifier type rather than changing types as it goes down the chain. Before DOT VALUE was always at the end of an access but now since it is just a general identifier this had to be done to prevent node.value.left